### PR TITLE
fix!: fix ``guilds`` property in client

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

Fixes an error occurring because of the cached guild object already having an HTTPClient

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): #664 
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [x] Bugfix

### CODE REVIEW

File: interactions/client.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def __init__(
     @property
     def guilds(self) -> List[Guild]:
         """Returns a list of guilds the bot is in."""
-        return [Guild(**_, _client=self._http) for _ in self._http.cache.self_guilds.view]
+        return [Guild(**_) for _ in self._http.cache.self_guilds.view]
```
Comment (i0bs) [inline interactions/client.py:105]: What are we ditching the `_client` declaration here for? Would it not be useful in terms of DX for a bot developer to be able to do something to a guild by easily reading off of this property?

```py
for guild in bot.guilds:
    if guild.name == "foo":
        await guild.method()
```

Comment (sillyElvy) [reply interactions/client.py:105]: it seems to be present already

I can't see any modification in cache, but ``'Guild' got multiple values for '_client'``

Comment (sillyElvy) [reply interactions/client.py:105]:
```py
Traceback (most recent call last):
  File "C:\Users\\Desktop\PycharmProjects\library\main.py", line 12, in on_ready
    print(bot.guilds)
  File "C:\Users\\Desktop\PycharmProjects\library\interactions\client.py", line 105, in guilds
    return [Guild(**_, _client=self._http) for _ in self._http.cache.self_guilds.view]
  File "C:\Users\\Desktop\PycharmProjects\library\interactions\client.py", line 105, in <listcomp>
    return [Guild(**_, _client=self._http) for _ in self._http.cache.self_guilds.view]
TypeError: interactions.api.models.guild.Guild() got multiple values for keyword argument '_client'
```

**Comment by i0bs** [comment]
In conjunction to `client.latency` which can be called upon prior to `client.start()`, I cannot access the `client.guilds` property method at all in an event listener in a main bot file. Trying this produces an `AttributeError`, see:

```py
@client.event
async def on_ready():
    print(len(client.guilds))
```

**Comment by sillyElvy** [comment]
for me it results in an empty list if not populated in the same event... how would it raise an attribute error?

**Comment by i0bs** [comment]
This is from the latest `unstable` commit:

image

**Comment by sillyElvy** [comment]
but that would mean the property is nonexistent... how's that possible?

**Comment by i0bs** [comment]
beats me i didn't write the code 😂

**Comment by sillyElvy** [comment]
> beats me i didn't write the code 😂

still doesn't make sense that a property, what is in the code, disappears on runtime?

**Comment by i0bs** [comment]
let me rephrase:

you're asking the wrong person why, i have zero clue why it's erroring. all i know is that it's broken and it needs to be fixed. pls fix

**Comment by sillyElvy** [comment]
Sorry I can't even start the bot because of `ImportError: cannot import name 'WebSocketClient' from partially initialized module 'interactions.api.gateway.client' (most likely due to a circular import) (C:\Users\edinv\Desktop\PycharmProjects\library\interactions\api\gateway\client.py)` 🙂

**Comment by Nanrech** [comment]
Relatable

**Comment by i0bs** [comment]
Latest unstable commits have fixed the circular import issue.

**Comment by sillyElvy** [comment]
```py
@bot.event
async def on_ready():
    print(bot.guilds)
    print(len(bot.guilds))
```

This results in:

grafik


I don't know what is causing the issue for you

**Comment by i0bs** [comment]
Seems that my env was messed up

File: interactions/bot.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def __init__(
     @property
     def guilds(self) -> List[Guild]:
         """Returns a list of guilds the bot is in."""
-        return [Guild(**_, _client=self._http) for _ in self._http.cache.self_guilds.view]
+        return [Guild(**_) for _ in self._http.cache.self_guilds.view]
```
Comment (i0bs) [inline interactions/bot.py:105]: How come we're ditching the client? What if bot developers want to call methods in a guild from here?

Comment (sillyElvy) [reply interactions/bot.py:105]: https://github.com/interactions-py/library/pull/670#discussion_r839827602

**Review by i0bs — APPROVED** [review]
